projects/core/src/i18n/translation-chunk.service.ts

```diff
@@ -3,25 +3,51 @@ import { I18nConfig } from './config/i18n-config';
 
 @Injectable()
 export class TranslationChunkService {
-  constructor(protected config: I18nConfig) {}
+  private duplicates: { [key: string]: string[] } = {};
+  private chunks: { [key: string]: string } = {};
+  constructor(protected config: I18nConfig) {
+    Object.keys(config.i18n.chunks).forEach(chunk => {
+      config.i18n.chunks[chunk].forEach(key => {
+        if (this.chunks.hasOwnProperty(key)) {
+          if (!this.duplicates[key]) {
+            this.duplicates[key] = [this.chunks[key]];
+          }
+          this.duplicates[key].push(chunk);
+        } else {
+          this.chunks[key] = chunk;
+        }
+      });
+    });
+    if (Object.keys(this.duplicates).length > 0 && !this.config.production) {
+      this.warnDuplicates(this.duplicates);
+    }
+  }
 
   protected readonly KEY_SEPARATOR = '.';
 
   getChunkNameForKey(key: string): string {
     const mainKey = (key || '').split(this.KEY_SEPARATOR)[0];
-    const chunk = this.getChunkFromConfig(mainKey);
+    const chunk = this.chunks && this.chunks[mainKey];
 
     if (!chunk) {
       return mainKey; // fallback to main key as a chunk
     }
     return chunk;
   }
 
-  private getChunkFromConfig(mainKey: string): string {
-    return (
-      this.config.i18n &&
-      this.config.i18n.chunks &&
-      this.config.i18n.chunks[mainKey]
+  private warnDuplicates(items: { [key: string]: string[] }): void {
+    const dupes: string[] = [];
+    Object.keys(items).forEach(key => {
+      dupes.push(
+        `* '${key}' key occurrences: ${items[key].join(', ')}. '${
+          this.chunks[key]
+        }.${key}' has been used.`
+      );
+    });
+    console.warn(
+      `Duplicated keys has been found in the 'defaultI18nConfig.i18n.chunks':\n${dupes.join(
```

Comment:
It not necessarily `defaultI18nConfig.i18n.chunks`. It can be also overwritten/extended in the `StorefrontModules`'s config in `app.module`. So can we name it simply `i18n chunks config`?